### PR TITLE
Rename unix.tcp to unix.stream, unix.udp to unix.dgram

### PR DESCRIPTION
**Pull request by enginix**
Changes for what we discussed in #193

### CODE REVIEW

**Comment by sathieu** [comment]
This is hard to follow. You're changing the API too often, and this breaks Prosody (See https://bugs.debian.org/852250).

**Comment by diegonehab** [comment]
The unix socket implementation has always been kind of a curiosity. Similar to the serial implementation. They are not even documented. In that sense, the API is not official. So I didn't feel very strongly about maintaining compatibility. I guess the best solution is to have a deprecation policy and have unix.tcp = unix.stream and unix.udp = unix.dgram temporarily. What do you say?

**Comment by sathieu** [comment]
What I need is socket.unix=socket.unix.tcp=socket.unix.stream, but is it possible?

**Comment by cbane** [comment]
I just opened a pull request (https://github.com/diegonehab/luasocket/pull/207) that adds backwards compatibility aliases and wrappers.

**Comment by diegonehab** [comment]
Sometimes the flexibility of Lua is a bit scary. Very nicely done.
